lib/background/insight.js

```diff
@@ -1,33 +1,68 @@
-function initGoogleAnalytics(trackingCode, doTrack) {
-  if (!doTrack) {
-    return;
-  }
+/* eslint-disable id-length */
+
+import querystring from 'querystring';
+import pkg from '../../package.json';
+
+let instance;
+const baseUrl = 'https://ssl.google-analytics.com/collect';
 
-  /* eslint-disable */
-  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
-  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
-  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
-  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
-  /* eslint-enable */
+const baseParams = ({ tid, cid }) => ({
+  // Check out the measurement protocol parameter reference for details.
+  // https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
+  v: 1,
+  tid, // Tracking ID
+  cid, // Anonymous identifier
+  aip: 1, // Anonymize IP,
+  an: pkg.name,
+  av: pkg.version,
+  ua: window.navigator.userAgent,
+  z: Date.now(),
+});
 
-  window.ga('create', 'UA-88792224-2', 'auto');
-  window.ga('set', 'anonymizeIp', true);
-  window.ga('set', 'checkProtocolTask', () => {}); // Removes failing protocol check. @see: http://stackoverflow.com/a/22152353/1958200
+function request(url, body) {
+  const xhr = new window.XMLHttpRequest();
+  xhr.open('POST', url);
+  xhr.send(querystring.stringify(body));
 }
 
-function trackEvent({ category, action, label, value }) {
-  if (!window.ga) {
-    return;
+const send = (requiredParams, params) => {
+  request(baseUrl, {
+    ...params,
+    ...baseParams(requiredParams),
+  });
+};
+
+class Insight {
+  constructor(trackingCode, clientId) {
+    this.tid = trackingCode;
+    this.cid = clientId;
+  }
+
+  track({ category, action, label, value }) {
+    const params = {
+      t: 'event',
+      ec: category,
+      ea: action,
+    };
+
+    if (label) params.el = label;
+    if (value) params.ev = value;
+
+    return send({
+      tid: this.tid,
+      cid: this.cid,
```

Comment:
The IP anonymization feature in Analytics sets the last octet of IPv4 user IP addresses and the last 80 bits of IPv6 addresses to zeros. For example `10.215.30.344` will be collected as `10.215.30.0`. By that it's "half" anonymized, because we still know the location of the local network switch, which is good enough.